crma_api_client/resources/dataset.py

```diff
@@ -16,6 +17,114 @@ class Dataset(BaseModel):
     url: str
 
 
+class DateType(Enum):
+    """The type of date field"""
+
+    date = "Date"
+    date_only = "DateOnly"
+    date_time = "DateTime"
+
+
+class XmdDateFields(BaseModel):
+    """Date fields in the extended metadata format
+
+    See https://developer.salesforce.com/docs/atlas.en-us.230.0.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_xmd_main.htm#XmdDateFieldRepresentation
+    """
+
+    day: str
+    epoch_day: str
+    epoch_second: str
+    fiscal_month: Optional[str]
+    fiscal_quarter: Optional[str]
+    fiscal_week: Optional[str]
+    fiscal_year: Optional[str]
+    full_field: str
+    hour: str
+    minute: str
+    month: str
+    quarter: str
+    second: str
+    week: str
+    year: str
+
+    @validator("fiscal_month", "fiscal_quarter", "fiscal_week", "fiscal_year")
+    def not_none(cls, v):
+        """The value should not be None"""
+        if v is None:
+            raise ValueError("Fiscal fields may not be None")
+        return v
```

Comment:
why is this validating that these fields are not none? they are listed as optional above

Reply:
`Optional[x]` is just shorthand for `Union[x, None]`. There's apparently no other way to say that the field isn't required but if it's there it can't be None

Reply:
ah gotcha